src/app_feup/lib/controller/library/library.dart

```diff
@@ -10,19 +11,25 @@ import 'package:uni/controller/library/library_utils.dart';
 import 'package:uni/controller/library/parser_library.dart';
 import 'package:uni/controller/library/parser_library_interface.dart';
 import 'package:uni/controller/local_storage/app_shared_preferences.dart';
+import 'package:uni/model/app_state.dart';
 import 'package:uni/model/entities/book.dart';
 import 'package:http/http.dart' as http;
 import 'package:uni/model/entities/search_filters.dart';
 
 import 'package:cookie_jar/cookie_jar.dart';
 import 'package:uni/model/entities/book_reservation.dart';
 
+// ignore: implementation_imports
+import 'package:redux/src/store.dart';
```

Comment:
```suggestion
import 'package:redux/redux.dart';
```
This solved it. You were trying to import from a private package instead of a public one. You can read about it here https://stackoverflow.com/questions/52925209/error-dont-import-implementation-files-from-another-package